packages/gestalt/src/ButtonSocial.tsx

```diff
@@ -0,0 +1,254 @@
+import { forwardRef, useImperativeHandle, useRef } from 'react';
+import classnames from 'classnames';
+import Box from './Box';
+import styles from './ButtonSocial.css';
+import VRButtonSocial from './ButtonSocial/VRButtonSocial';
+import { useColorScheme } from './contexts/ColorSchemeProvider';
+import { useDefaultLabelContext } from './contexts/DefaultLabelProvider';
+import Flex from './Flex';
+import focusStyles from './Focus.css';
+import Icon from './Icon';
+import touchableStyles from './TapArea.css';
+import Text from './Text';
+import useFocusVisible from './useFocusVisible';
+import useInExperiment from './useInExperiment';
+
+const TYPE_OPTIONS = {
+  'login': 'Login with',
+  'continue': 'Continue with',
+  'signup': 'Sign up with',
+};
+
+const SERVICES_OPTIONS = {
+  'apple': 'Apple',
+  'facebook': 'Facebook',
+  'google': 'Google',
+  'email': 'Email',
+};
+
+type ButtonProps = {
+  /**
+   * Available for testing purposes, if needed. Consider [better queries](https://testing-library.com/docs/queries/about/#priority) before using this prop.
+   */
+  dataTestId?: string;
+  /**
+   * Callback invoked when the user clicks (press and release) on Button with the mouse or keyboard.
+   */
+  onClick?: (arg1: {
+    event: React.MouseEvent<HTMLButtonElement> | React.KeyboardEvent<HTMLButtonElement>;
+  }) => void;
+  /**
+   * Text to render inside the ButtonSocial to convey the function and purpose of the ButtonSocial.
+   */
+  type: 'login' | 'continue' | 'signup';
+  /**
+   * Text to render inside the ButtonSocial to convey the function and purpose of the ButtonSocial.
+   */
+  service: 'apple' | 'facebook' | 'google' | 'email';
+};
+
+/**
+ * [ButtonSocial](https://gestalt.pinterest.systems/buttonsocial) should be used only to enable users to sign-up or sign-in to Pinterest using other trusted services.
+ *
+ * ![ButtonSocial light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/ButtonSocial.spec.ts-snapshots/ButtonSocial-chromium-darwin.png)
+ * ![ButtonSocial dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/ButtonSocial-dark.spec.ts-snapshots/ButtonSocial-dark-chromium-darwin.png)
+ */
+
+const ButtonSocialWithForwardRef = forwardRef<HTMLAnchorElement, ButtonProps>(function ButtonSocial(
+  { dataTestId, onClick, type, service }: ButtonProps,
```

Comment:
`  { dataTestId, onClick, type, service }: ButtonProps,
`

just 

`  { dataTestId, onClick, type, service }
`

: ButtonProps, is redundant